neurovault/apps/statmaps/views.py

```diff
@@ -447,23 +444,40 @@ def add_image_for_neurosynth(request):
         priv_token = generate_url_token()
         temp_collection = Collection(name=temp_collection_name,
                                      owner=request.user,
-                                     private=False,
+                                     private=is_private,
                                      private_token=priv_token)
         temp_collection.save()
     image = StatisticMap(collection=temp_collection)
+    redirect = redirect_url+"/?neurovault=%s-%s" % (
+        temp_collection.private_token,image.id)
     if request.method == "POST":
-        form = SimplifiedStatisticMapForm(request.POST, request.FILES, instance=image, user=request.user)
+        if app == "neurosynth":
+            form = SimplifiedStatisticMapForm(request.POST, request.FILES, instance=image, user=request.user)
```

Comment:
Instead of passing "app" argument and doing this conditional could you just pass the form class (for example `SimplifiedStatisticMapForm`)?